### ISSUE TITLE
Upgrade third-party dependency Eigen to version 3.4

### ISSUE THREAD

**Issue by severinstrobl** [severinstrobl/overlap#25]
Recently Eigen 3.4.0 was [released](https://eigen.tuxfamily.org/index.php?title=3.4), marking the first major release in a while. The overlap library should be tested with this release and if everything works out the Eigen sub-module should be updated accordingly.

**Comment by severinstrobl** [comment]
The switch to Eigen 3.4.0 is done in #48 and included in the development branch. Once the development branch is merged into the mainline, this issue can be closed.
